src/middleware/withDelegationsStorage.types.ts

```diff
@@ -0,0 +1,48 @@
+import * as Ucanto from '@ucanto/interface'
+import { Environment as MiddlewareEnvironment, Context as MiddlewareContext } from '@web3-storage/gateway-lib'
+import { KVNamespace } from '@cloudflare/workers-types'
+import { SpaceDID } from '@web3-storage/capabilities/types'
+import { Failure } from '@ucanto/core'
+import { GatewayIdentityContext } from './withGatewayIdentity.types.js'
+import { LocatorContext } from './withLocator.types.js'
+
+export interface Environment extends MiddlewareEnvironment {
+  CONTENT_SERVE_DELEGATIONS_STORE: KVNamespace
+  FF_DELEGATIONS_STORAGE_ENABLED: string
+}
+
+export class DelegationFailure extends Failure {
+  get name() {
+    return /** @type {const} */ ('DelegationFailure')
+  }
+}
+
+export interface DelegationsStorageContext
+  extends MiddlewareContext,
+  GatewayIdentityContext {
+  delegationsStorage: DelegationsStorage
+}
+
+export interface DelegationsStorage {
+  /**
+   * Finds the delegation proofs for the given space
+   * 
+   * @param {import('@web3-storage/capabilities/types').SpaceDID} space 
+   * @returns {Promise<Ucanto.Result<Ucanto.Delegation<Ucanto.Capabilities>, Ucanto.Failure>>}
```

Comment:
🙏 can we be more specific with the failure type? i.e. `DelegationNotFound | StoreOperationFailed`